### PR TITLE
drivers: bluetooth: silabs: move code from module to main tree

### PR DESCRIPTION
**Pull request by jrintaha**
The code that handles passing events to and from the controller resided in the hal_silabs module. This is an integral part of the HCI driver that it can't work without, and logically belongs in the driver.

### CODE REVIEW

File: drivers/bluetooth/hci/slz_hci.c [hunk 1]
```diff
@@ -120,13 +131,26 @@ static int slz_bt_send(const struct device *dev, struct net_buf *buf)
 	return rv;
 }
 
+
```
Comment (jhedberg) [inline drivers/bluetooth/hci/slz_hci.c:134]: same here

**Comment by jhedberg** [comment]
@jrintaha thanks! Looks generally fine, just two small whitespace comments.

File: drivers/bluetooth/hci/slz_hci.c [hunk 2]
```diff
@@ -34,6 +33,18 @@ struct hci_data {
 static K_KERNEL_STACK_DEFINE(slz_ll_stack, SL_BT_SILABS_LL_STACK_SIZE);
 static struct k_thread slz_ll_thread;
 
+/* Semaphore for Link Layer */
+K_SEM_DEFINE(slz_ll_sem, 0, 1);
+
+/* Events mask for Link Layer */
+static atomic_t sli_btctrl_events;
+
+
```
[the remark this answers is not in the document]
Comment (jrintaha) [reply drivers/bluetooth/hci/slz_hci.c]: Gotcha. Curiously, checkpatch.pl did not complain about these. It did catch other style issues that I fixed before pushing.

**Review by jhedberg — DISMISSED** [review]

**Review by jerome-pouiller — COMMENTED** [review]
Usually, in the commit title, `silabs` is written in lowercase (because it is our prefix in Device Tree bindings). The vendor also appears after the type of driver. And finally, it seems the bluetooth subsystem uses plain `bluetooth` name rather than `bt` in their commit log.

So, I think the title should start with something like `drivers: bluetooth: silabs:` (or `driver: Bluetooth: hci: silabs:`).

Otherwise, LGFM.

**Review by jhedberg — APPROVED** [review]

**Review by jerome-pouiller — APPROVED** [review]

**Pull request merged**